ReactApp/src/sections/popUps/QuestionPhotoView.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import { requestLatexServer } from '../../utils/rest-api-call';
+import { pdfjs, Document, Page } from 'react-pdf';
+import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
+import 'react-pdf/dist/esm/Page/TextLayer.css';
+
+import pdfWorker from 'pdfjs-dist/build/pdf.worker.min.mjs';
+import { Typography } from '@mui/material';
+import ErrorMessage from '../../components/errorMessage';
+
+pdfjs.GlobalWorkerOptions.workerSrc = new URL(
+  pdfWorker,
+  import.meta.url,
+).toString();
+
+const QuestionPhotoView = ({ content, type }) => {
+  const [pdfUrl, setPdfUrl] = useState('');
+  const [error, setError] = useState(null);
+  const [retries, setRetries] = useState(0);
+  const MAX_RETRIES = 3;
+
+  const retryMechanism = (message) => {
+    if (retries < MAX_RETRIES) {
+      setRetries(retries + 1);
+    }
+    else {
+      setError(message);
+    }
+    console.error(message);
+  };
+
+  useEffect(() => {
+    const fetchPDF = async () => {
+      try {
+        const response = await requestLatexServer(type, { content });
+        if (!response.ok) {
+          retryMechanism("Failed to load PDF file");
+        }
+        const blob = await response.blob();
```

Comment:
For my knowladge, I saw a lot of blob, but never knew what that means.
Can you please explain me?